### ISSUE TITLE
Solve does not support saveat vectors for ensemble problems

### ISSUE THREAD

**Issue by toollu** [SciML/DifferentialEquations.jl#671]
In the contest of "abusing" ensembles to solve an inverse problem on i datasets (with different initial conditions, time steps and tends) it would be cool if 

```
function prob_func(prob,i,repeat)
  ODEProblem(prob.f,initial_conditions[i],tspans[i],prob.p)
end

monte_prob = MonteCarloProblem(prob,prob_func=prob_func)

sim = solve(monte_prob,trajectories=i,saveat = saveat_vec)
```

would work (where saveat_vec is of type `Vector{StepRangeLen}`).
My current solution is to just have `saveat=tstep` and interpolate the fitting data to those tsteps.

**Comment by ChrisRackauckas** [comment]
```julia
function prob_func(prob,i,repeat)
  ODEProblem(prob.f,initial_conditions[i],tspans[i],prob.p,saveat = saveat_vec)
end
```

should be sufficient.

**Comment by toollu** [comment]
Thank you. This suggestion with adding an index did indeed work:

 ```Julia
  function prob_func(prob,i,repeat)
    ODEProblem(prob.f,initial_conditions[i],tspans[i],prob.p,saveat = saveat_vec[i])
  end
```

Somewhat related issue. Is it possible to do something like 

 ```Julia
  function prob_func(prob,i,repeat)
    ODEProblem(prob.f,initial_conditions[i],tspans[i], p=[prob.p; not_fitted_parameters[i]], saveat = saveat_vec[i])
  end
```

to define a subset of parameters that differ in the ensemble but are not fitted when solve() is passed to an optimizer?
Currently I see no solution for this other than redefining the Ensemble problem at every optimisation solution.

**Comment by ChrisRackauckas** [comment]
Yes, that'll work.

**Comment by ChrisRackauckas** [comment]
Or you can enclose not-changed parameters into the function itself.

**Comment by toollu** [comment]
> Yes, that'll work.

Unfortunately not (unless I‘m missing something) `solve(ensemble_problem,solver(),p=theta)` results in theta overwriting of everything defined in `prob_func` and not only the `prob.p`. Defining everything in the function itself also does not work, since the ensemble consist of functions that have one subset of parameters for construction of different ensemble members. Those members than share another subset of parameters I want to solve an inverse problem on. Another idea I had was extracting a vector of prob.f[i] from a wrapper  around the ode construction and pass some kind of dummy ODEProblem to the ensemble problem constructor. Couldn’t make it work though so far (and does feel a little bit clunky for how elegant the  rest of the approach is).

Btw thanks for your efforts! I‘ll be happy to provide & write a toy example of this specific problem to the documentation (once I can make it work) since I imagine I‘m not the only one that wants to fit time series with different properties defined by a set of different parameters but with the same underlying mathematical equations described by another set of parameters (the ones not known a priori).

**Comment by ChrisRackauckas** [comment]
>Unfortunately not (unless I‘m missing something) solve(ensemble_problem,solver(),p=theta) results in theta overwriting of everything defined in prob_func and not only the prob.p

I wouldn't recommend doing the `solve` version if you're using a `prob_func`: all it's doing is `remake` under the hood.

>Defining everything in the function itself also does not work, since the ensemble consist of functions that have one subset of parameters for construction of different ensemble members

If you're only differentiating a subset, I would recommend using a closure over the function anyways to decrease the computational cost.

**Comment by toollu** [comment]
Thank you Chris.
The solution I have now is:

```Julia
function sysdefine(u0, tspan,k)
  return ODEProblem((u,odesyspars,t) -> odesys(u,odesyspars,t, k),u0,tspan)
end
```

Where `k` is the additional parameter to generate the different ODE Problems then munched together in a loss function to be trained on another, common set of parameters to be optimised. Still having issues with Autodiff/Zygote. As it is  not related however so I closed the issue.

**Comment by ChrisRackauckas** [comment]
Yup that's what I was expecting here (sans autodiff issues). Cheers.
